### DISCUSSION TITLE
A function call/getIfExists

### DISCUSSION THREAD

**Discussion by kaatinga** [jellydator/ttlcache#126] [Ideas]
I really love ttlcache and use it in every my project.

Recently I implemented a basic controller logic and fount it convenient to use ttlacache again yet I found one feature which would bring atomic operations to the cache only in case the cache item exists.

I will elaborate. The Get() method allows to fetch the cache item, yet, it also runs the loader if it is used in the cache. But what I need is to distinguish the case when I want to get the item when it exists already and when I want to get it and create if it does not exist yet. A good option for me is to use a callback which would be executed for the cache item only in case it exists already.

Now I have to use the Has() method and then Get() method. But between them item can be deleted and Get() method will create the item again what I do not need.

**Comment by davseby** [comment]
`Get` accepts options, you can either create a cache with a loader and then disable it for the specific `Get` calls or you can do it the other way around - create a cache without a loader and then supply it via `Get` parameters.

Wouldn't this solve the issue?

**Reply by kaatinga** [reply]
Actually I need to apply changes to the item, so the simplest option here is to use an object in the cache that has mutex and public methods to work with the object which would be thread-safe

My idea was to add a helper to work with the cache item atomically
